src/main/java/teammates/ui/controller/FeedbackSubmissionEditSaveAction.java

```diff
@@ -234,6 +252,213 @@ protected ActionResult execute() throws EntityDoesNotExistException {
+        } else {
+            feedbackResponseComment.setVisibilitySettingsForInstructorComment(showCommentTo, showGiverNameTo);
+        }
+
+        FeedbackResponseCommentAttributes updatedComment = null;
+        try {
+            updatedComment = logic.updateFeedbackResponseComment(feedbackResponseComment);
+            //TODO: move putDocument to task queue
+            logic.putDocument(updatedComment);
+            statusToAdmin += "FeedbackSubmissionEditSaveAction:<br>"
+                                     + "Editing feedback response comment: " + feedbackResponseComment.getId() + "<br>"
+                                     + "in course/feedback session: " + feedbackResponseComment.courseId + "/"
+                                     + feedbackResponseComment.feedbackSessionName + "<br>"
+                                     + "by: " + feedbackResponseComment.commentGiver + "<br>"
+                                     + "comment text: " + feedbackResponseComment.commentText.getValue();
+        } catch (InvalidParametersException e) {
+            setStatusForException(e);
+            isError = true;
+        } catch (EntityDoesNotExistException e) {
+            isError = true;
+        }
+    }
+
+    /**
+     * Saves comments which were newly added to feedback response.
+     * @param commentsToAddText key: comment index, value: comment text
+     * @param responseGiverMapForComments key: comment index, value: response giver
+     * @param responseRecipientMapForComments key: comment index, value: response recipient
+     * @param questionIdsForComments key: comment index, value: question id
+     */
+    private void saveResponsesComments(Map<String, String> commentsToAddText,
+                                       Map<String, String> responseGiverMapForComments,
+                                       Map<String, String> responseRecipientMapForComments,
+                                       Map<String, String> questionIdsForComments) throws EntityDoesNotExistException {
+        for (String commentIndex : commentsToAddText.keySet()) {
+            String responseGiver = responseGiverMapForComments.get(commentIndex);
+            String responseRecipient = responseRecipientMapForComments.get(commentIndex);
+            String commentText = commentsToAddText.get(commentIndex);
+            String questionId = questionIdsForComments.get(commentIndex);
+            FeedbackResponseAttributes responseToAddComment =
+                    logic.getFeedbackResponse(questionId, responseGiver, responseRecipient);
+            String commentGiverTypeString =
+                    getRequestParamValue(Const.ParamsNames.COMMENT_GIVER_TYPE + commentIndex);
+            String showCommentTo = null;
+            String showGiverNameTo = null;
+            // Only instructor can set visibility settings for his/her comments.
+            if (this instanceof InstructorFeedbackSubmissionEditSaveAction) {
```

Comment:
I think there is a misunderstanding here.

`InstructorFeedbackSubmissionEditSaveAction` is for instructor to submit feedback. That is nothing to do with the comment added by the instructor. So instructor in this case is the feedback participant.

For example, if I am evaluator and I am going to give comment based on your feedback to others, the `isCommentFromFeedbackParticipant` should be `false` as I am a third person. If you are a instructor and your have given feedback to other instructors. In addition, you have given a comment to elaborate your choice, `isCommentFromFeedbackParticipant` should be `true` in this case as you are feedback participant. Ask me more on Slack if you have any question.